cas-server-core/src/main/java/org/jasig/cas/CentralAuthenticationServiceImpl.java

```diff
@@ -279,14 +279,14 @@ public ServiceTicket grantServiceTicket(
         getAuthenticationSatisfiedByPolicy(ticketGrantingTicket, new ServiceContext(service, registeredService));
 
         final String uniqueTicketIdGenKey = service.getClass().getName();
-        if (!this.uniqueTicketIdGeneratorsForService.containsKey(uniqueTicketIdGenKey)) {
-            logger.warn("Cannot create service ticket because the key [{}] for service [{}] is not linked to a ticket id generator",
-                    uniqueTicketIdGenKey, service.getId());
-            throw new UnauthorizedSsoServiceException();
-        }
-        
-        final UniqueTicketIdGenerator serviceTicketUniqueTicketIdGenerator =
+        logger.debug("Looking up service ticket id generator for [{}]", uniqueTicketIdGenKey);

```

Comment:
Tough to say! My experiencing in arriving at this pull has thus far been dealing with some specialized deployments that merited parsing request parameters into a separate Service type. It still feels like something of an override would be in order, if only needed to bypass the default.  On average, I'd say chances of someone creating a new service are far more likely than requiring a particular type of id generator. 

The other option might be to not add a default, but locate the generator based on an inheritance hierarchy rather than exact matches but that would obviously require a particular chain of components which may or may not make sense for a deployment.